pkg/streamingpromql/operators/selectors/range_vector_selector.go

```diff
@@ -82,17 +95,70 @@ func (m *RangeVectorSelector) NextStepSamples(ctx context.Context) (*types.Range
 	// Apply offset after adjusting for timestamp from @ modifier.
 	rangeEnd = rangeEnd - m.Selector.Offset
 	rangeStart := rangeEnd - m.rangeMilliseconds
+
+	// Take a copy of the original range - the smoothed/anchored modifiers will change these
+	originalRangeStart := rangeStart
+	originalRangeEnd := rangeEnd
+
+	// When the smoothed/anchored modifiers are used, the selector (fillBuffer) will return a wider range of points.
+	// The range boundaries are modified accordingly so that we do not discard these extended points.
```

Comment:
[nit] The past tense here makes it sound like the range boundaries have already been adjusted, but we're about to adjust them in the `if` below:

```suggestion
	// Modify the range boundaries accordingly so that we do not discard these extended points.
```